webhooks/models.py

```diff
@@ -0,0 +1,170 @@
+    updated_at: str
+
+
+def to_connections_model(item: dict) -> ConnectionsHook:
+    if item["connection_protocol"] == "didexchange/1.0":
+        item["connection_id"] = "0023-" + item["connection_id"]
+    elif item["connection_protocol"] == "connections/1.0":
+        item["connection_id"] = "0016-" + item["connection_id"]
+    try:
+        item = ConnectionsHook(**item)
+    except ValidationError:
+        pass
+    return item
+
+
+class BasicMessagesHook(HookBase):
+    connection_id: str
+    content: str
+    message_id: str
+    sent_time: str
+    state: Literal["received"]
+
+
+class ProofsHook(HookBase):
+    connection_id: Optional[str]
+    created_at: str
+    initiator: Literal["self", "external"]
+    role: Literal["prover", "verifier"]
+    protocol_version: Literal["v1", "v2"]
+    presentation_exchange_id: str
+    presentation_request: Optional[dict]
+    presentation_request_dict: Optional[dict]
+    state: Literal[
+        "proposal-sent",
+        "proposal-received",
+        "request-sent",
+        "request-received",
+        "presentation-sent",
+        "presentation-received",
+        "verified",
+    ]
+    thread_id: str
+    trace: bool
+    updated_at: str
+
+
+def to_proof_hook_model(item: dict) -> ProofsHook:
+    if "presentation_exchange_id" in item:
+        item["protocol_version"] = "v1"
+        item["presentation_exchange_id"] = "v1-" + item["presentation_exchange_id"]
+        try:
+            item["state"] = item["state"].replace("_", "-")
+        except KeyError:
+            pass
+        item["presentation_exchange_id"] = "v1-" + item["presentation_exchange_id"]
+        item = ProofsHook(**item)
+    elif "pres_ex_id" in item:
+        item["pres_ex_id"] = "v2-" + item["pres_ex_id"]
+        item["presentation_exchange_id"] = item.pop("pres_ex_id")
+        item["protocol_version"] = "v2"
+        item["presentation_request"] = item.pop("pres_request")
+        item = ProofsHook(**item)
+    return item
+
+
+class CredentialsHooks(HookBase):
+    auto_issue: Optional[bool]
+    auto_offer: Optional[bool]
+    auto_remove: Optional[bool]
+    connection_id: Optional[str]
+    created_at: str
+    protocol_version: Literal["v1", "v2"]
+    initiator: Literal["self", "external"]
+    role: Literal["issuer", "holder"]
+    state: Optional[
+        Literal[
+            "proposal-sent",
+            "proposal-received",
+            "offer-sent",
+            "offer-received",
+            "request-sent",
+            "request-received",
+            "credential-issued",
+            "credential-received",
+            "done",
+        ]
+    ]
+    thread_id: Optional[str]
+    trace: bool
+    updated_at: str
+    credential_exchange_id: Optional[str]
+    credential_definition_id: Optional[str]
+    credential_proposal: Optional[dict]
+    credential_offer: Optional[dict]
+
+
+def to_credentential_hook_model(item: dict) -> CredentialsHooks:
```

Comment:
Would be good to reuse the app logic here. We already have methods that transform v1 and v2 ACA-Py credential / proof records to the CloudAPI model.

Reply:
yes. However, as far as I know I'd still have to write that code out here because sharing between the two apps/containers isn't really doable nicely - but let me know if you have a good idea though

Reply:
I'll have a go and transform the models to be like what we have in the API already where applicable.

Reply:
I've made the models shared between the two apps now.

For the validation logic, as much as I agree with this being better shared also, I am in favour of doing this as an improvement in a separate PR. That is for the cloudapi app has this logic partially as class (private) methods and further validates this against aries-cloudcontroller types that are being returned. The webhooks do this against a bytestream from Redis instead. Having had a look at it, I think this a slightly delicate matter that requires some effort that lies beyond the scope of this PR imho. Hence, I'd do it in a different PR.